Automattic/wp-calypso
# WIP Pattern Assembler: Render block patterns on calypso side

#### Proposed Changes

* Use the `pattern-assembler/client-side-render` flag to control the client-side render
* Introduce the `@automattic/blocks-renderer` package to support rendering the block patterns on the calypso side. The main concept is as followed:
  * `BlocksProvider`: Fetch the block settings from API and store it in the store of the `block-editor`
  * `BlocksRenderer`: Render the HTML and generated styles of the block pattern. The HTML of the block pattern is from the `/block-patterns/render` endpoint.
  * `PatternsRendererProvider`: The provider to batch query the rendered patterns from the server side
  * `PatternsRenderer`: Render the patterns with the given pattern ids
* This PR only uses client-side rendering on the sidebar. Will continue to apply the client-side rendering on the large preview.

See pbxlJb-30c-p2 for more context

TODOs

- [ ] Render site logo placeholder
- [ ] Scroll to the specified pattern when the selected pattern set is changed
- [ ] Maybe removing animation

https://user-images.githubusercontent.com/13596067/203761687-0ebe8afc-8174-45ec-ab24-769af15f7534.mov

#### Testing Instructions



* Apply D86835-code to your sandbox
* Go to `/setup?siteSlug=<your_site>`
* Click the "Continue" button until you land on the Design Picker step
* Click the "Get started" button below "Start with a blank canvas" to enter the Pattern Assembler step
* On the Pattern Assembler step, verify all of the patterns shown in the sidebar are rendered correctly

#### Pre-merge Checklist





Related to https://github.com/Automattic/wp-calypso/pull/67187, https://github.com/Automattic/wp-calypso/issues/67076
